Description: Fix segfault or infinite loop in delscreen()
 Patch taken from upstream patchlevel 20100501.
Bug-Debian: http://bugs.debian.org/597175
Author: Thomas Dickey <dickey@his.com>
Last-Update: 2010-09-17
---
 ncurses/base/lib_set_term.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ncurses/base/lib_set_term.c
+++ b/ncurses/base/lib_set_term.c
@@ -117,7 +117,7 @@
     for (each_screen(temp)) {
 	if (temp == sp) {
 	    if (last)
-		last = sp->_next_screen;
+		last->_next_screen = sp->_next_screen;
 	    else
 		_nc_screen_chain = sp->_next_screen;
 	    result = TRUE;
